pimd/pim_nht.c

```diff
@@ -171,15 +171,39 @@ void pim_delete_tracked_nexthop(struct pim_instance *pim, struct prefix *addr,
 	struct pim_nexthop_cache *pnc = NULL;
 	struct pim_nexthop_cache lookup;
 	struct zclient *zclient = NULL;
+	struct listnode *upnode = NULL;
+	struct pim_upstream *upstream = NULL;
 
 	zclient = pim_zebra_zclient_get();
 
 	/* Remove from RPF hash if it is the last entry */
 	lookup.rpf.rpf_addr = *addr;
 	pnc = hash_lookup(pim->rpf_hash, &lookup);
 	if (pnc) {
-		if (rp)
+		if (rp) {
```

Comment:
this code doesn't belong here. instead when an rp is deleted the upstream entry needs to be appropriately updated -
1. rpf cleared
2. nht removed from the rp

please move all this into a separate function and call it from pim_rp_del.